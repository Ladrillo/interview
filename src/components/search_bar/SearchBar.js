import React, { Component } from 'react';
import { Observable } from 'rxjs/Observable';
import 'rxjs/add/observable/fromEvent';
import 'rxjs/add/operator/debounceTime';
import 'rxjs/add/operator/merge';
import './SearchBar.css';


export default class SearchBar extends Component {
    constructor() {
        super();
        this.state = {
            searchTerm: ''
        };
        this.onChange = this.onChange.bind(this);
    }

    componentDidMount() {
        // we only want to AJAX  after 200 milliseconds of 'radio silence'
        const input = document.querySelector('.search-bar-input');
        const { searchHandler } = this.props;

        const stream = Observable.fromEvent(input, 'input');

        stream
            .debounceTime(200)
            .subscribe(e => searchHandler(this.state.searchTerm));
    }

    onChange(event) {
        const searchTerm = event.target.value;
        this.setState({ searchTerm });
    }

    render() {
        return (
            <div className='search-bar'>
                <input
                    className   = 'search-bar-input'
                    placeholder = 'Search Your Destiny'
                    value       = {this.state.searchTerm}
                    onChange    = {this.onChange}
                />
            </div>
        );
    }
}
