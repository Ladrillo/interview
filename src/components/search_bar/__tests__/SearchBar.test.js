import SearchBar from '../SearchBar';
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';


describe('SearchBar', () => {
    it('renders correctly', () => {
        const rendered = renderer.create(
            <SearchBar debounce={false} />
        );

        expect(rendered.toJSON()).toMatchSnapshot();
    });
});
