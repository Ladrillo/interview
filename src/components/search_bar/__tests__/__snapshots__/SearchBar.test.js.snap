exports[`SearchBar renders correctly 1`] = `
<div
  className="search-bar">
  <input
    className="search-bar-input"
    onChange={[Function]}
    placeholder="Search Your Destiny"
    value="" />
</div>
`;
